src/unity/lib/extensions/ml_model.hpp

```diff
@@ -122,7 +122,7 @@ class EXPORT ml_model_base: public model_base {
    *
    * This is the function that the list_fields should call in python.
    */
-  std::vector<std::string> list_fields();
+  std::vector<std::string> list_fields() const;
```

Comment:
We can't actually change this to const because the cpp_ipc base class that calls these functions doesn't have the ability to resolve const  vs. non-const with the current RPC stuff.  Once that class is separated out as a base class, we can change all of these (and other methods like name()) to const.  

Let's make sure that we keep the changes in the current PRs relevant to the task, both for tracking changes and to prevent irrelevant issues from holding up the task at hand.

Reply:
Good catch. I've reverted that change.

Without care, codebases tend to decay over time. With each PR, I think it's important to leave the code base in a better condition than you've found it. This is the best method I know to combat decay.